src/EncompassRest/ResourceLocks/ResourceLocks.cs

```diff
@@ -0,0 +1,132 @@
+            Preconditions.NotNullOrEmpty(lockType, nameof(lockType));
+
+            ResourceLock resourceLock = new ResourceLock();
+            resourceLock.Resource.EntityId = resourceId;
+            resourceLock.Resource.EntityType = resourceType;
+            resourceLock.LockType = lockType;
+
+            return LockResourceAsync(resourceLock, force, false, cancellationToken);
+        }
+
+        public Task<string> LockResourceRawAsync(string resourceLock, string queryString, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceLock, nameof(resourceLock));
+
+            return PostAsync(null, queryString, new JsonStringContent(resourceLock), nameof(LockResourceRawAsync), null, cancellationToken, ReadAsStringElseLocationFunc);
+        }
+
+        public Task<bool> UnlockResourceAsync(string lockId, string resourceId, EntityType resourceType, CancellationToken cancellationToken = default) =>
+            UnlockResourceAsync(lockId, resourceId, resourceType, false, cancellationToken);
+        public Task<bool> UnlockResourceAsync(string lockId, string resourceId, EntityType resourceType, bool force, CancellationToken cancellationToken = default) =>
+            UnlockResourceAsync(lockId, resourceId, resourceType.ToString(), force, cancellationToken);
+        public Task<bool> UnlockResourceAsync(string lockId, string resourceId, string resourceType, CancellationToken cancellationToken = default) =>
+            UnlockResourceAsync(lockId, resourceId, resourceType.ToString(), false, cancellationToken);
+        public Task<bool> UnlockResourceAsync(string lockId, string resourceId, string resourceType, bool force, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceId, nameof(resourceId));
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+            if (force) queryParameters.Add(new QueryParameter("force", "true"));
+
+            return DeleteAsync(lockId, queryParameters.ToString(), cancellationToken);
+        }
+        public Task<bool> UnlockResourceAsync(ResourceLock resourceLock, CancellationToken cancellationToken = default) => UnlockResourceAsync(resourceLock, false, cancellationToken);
+        public Task<bool> UnlockResourceAsync(ResourceLock resourceLock, bool force, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNull(resourceLock, nameof(resourceLock));
+
+            return UnlockResourceAsync(resourceLock.Id, resourceLock.Id, resourceLock.LockType, force, cancellationToken);
```

Comment:
this should be `return UnlockResourceAsync(resourceLock.Id, resourceLock.Resource.EntityId, resourceLock.Resource.EntityType, force, cancellationToken);`